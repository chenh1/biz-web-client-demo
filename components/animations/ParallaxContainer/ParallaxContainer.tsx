'use client'

import React from 'react';
import { useVerticalParallax } from '../../hooks/useVerticalParallax';

type ParallaxContainer = {
  children: string | JSX.Element | JSX.Element[], 
  className?: string, 
  parallaxAmount?: number,
  offsetAmt?: number,
  animationSpeedClass?: `duration-${number}`
}

export const ParallaxContainer: React.FC<ParallaxContainer> = ({ className = "", children, parallaxAmount = 0.5, offsetAmt = 0, animationSpeedClass = 'duration-300' }) => {
  const { elementRef } = useVerticalParallax({ parallaxAmount, offsetAmt });

  return (
    <div ref={elementRef} className={`${animationSpeedClass} ${className}`}>
      {children}
    </div>
  );
};