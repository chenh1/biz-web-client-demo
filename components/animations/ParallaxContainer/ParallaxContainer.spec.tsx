import { ParallaxContainer } from "./ParallaxContainer";
import { render, waitFor } from '@testing-library/react';
import { JSDOM } from "jsdom"
const dom = new JSDOM()
global.document = dom.window.document
global.window = dom.window

it('slides with a parallax effect', async () => {
  const { getByTestId, asFragment } = render(
    <ParallaxContainer>
      <p>Parallax</p>
    </ParallaxContainer>
  );

  await waitFor(() => {
    expect(asFragment()).toMatchSnapshot();
  });
});