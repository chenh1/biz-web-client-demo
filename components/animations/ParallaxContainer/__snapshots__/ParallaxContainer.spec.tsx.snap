// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`slides with a parallax effect 1`] = `
<DocumentFragment>
  <div
    class="duration-300 "
  >
    <p>
      Parallax
    </p>
  </div>
</DocumentFragment>
`;
